runtime/binding-http-kafka/src/main/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/config/HttpKafkaOptionsConfigAdapter.java

```diff
@@ -111,45 +113,40 @@ public JsonObject adaptToJson(
     public OptionsConfig adaptFromJson(
         JsonObject object)
     {
-        HttpKafkaIdempotencyConfig newIdempotency = IDEMPOTENCY_DEFAULT;
+        HttpKafkaCorrelationConfigBuilder<HttpKafkaCorrelationConfig> correlationConfig = HttpKafkaCorrelationConfig.builder();
+        HttpKafkaIdempotencyConfigBuilder<HttpKafkaIdempotencyConfig> idempotencyConfig = HttpKafkaIdempotencyConfig.builder();
 
         if (object.containsKey(IDEMPOTENCY_NAME))
         {
-            String8FW newIdempotencyKey = IDEMPOTENCY_HEADER_DEFAULT;
-
             JsonObject idempotency = object.getJsonObject(IDEMPOTENCY_NAME);
             if (idempotency.containsKey(IDEMPOTENCY_HEADER_NAME))
             {
-                newIdempotencyKey = new String8FW(idempotency.getString(IDEMPOTENCY_HEADER_NAME));
+                idempotencyConfig.header(idempotency.getString(IDEMPOTENCY_HEADER_NAME));
             }
-
-            newIdempotency = new HttpKafkaIdempotencyConfig(newIdempotencyKey);
         }
 
-        HttpKafkaCorrelationConfig newCorrelation = CORRELATION_DEFAULT;
         if (object.containsKey(CORRELATION_NAME))
         {
             JsonObject correlation = object.getJsonObject(CORRELATION_NAME);
             if (correlation.containsKey(CORRELATION_HEADERS_NAME))
             {
                 JsonObject headers = correlation.getJsonObject(CORRELATION_HEADERS_NAME);
 
-                String16FW newReplyTo = CORRELATION_HEADERS_REPLY_TO_DEFAULT;
                 if (headers.containsKey(CORRELATION_HEADERS_REPLY_TO_NAME))
                 {
-                    newReplyTo = new String16FW(headers.getString(CORRELATION_HEADERS_REPLY_TO_NAME));
+                    correlationConfig.replyTo(headers.getString(CORRELATION_HEADERS_REPLY_TO_NAME));
                 }
 
-                String16FW newCorrelationId = CORRELATION_HEADERS_CORRELATION_ID_DEFAULT;
                 if (headers.containsKey(CORRELATION_HEADERS_CORRELATION_ID_NAME))
                 {
-                    newCorrelationId = new String16FW(headers.getString(CORRELATION_HEADERS_CORRELATION_ID_NAME));
+                    correlationConfig.correlationId(headers.getString(CORRELATION_HEADERS_CORRELATION_ID_NAME));
                 }
-
-                newCorrelation = new HttpKafkaCorrelationConfig(newReplyTo, newCorrelationId);
             }
         }
 
-        return new HttpKafkaOptionsConfig(newIdempotency, newCorrelation);
+        return HttpKafkaOptionsConfig.builder()
+            .idempotency(idempotencyConfig.build())
+            .correlation(correlationConfig.build())
+            .build();
```

Comment:
Move the `options` builder up and add methods like `.idempotency()` that return a `HttpKafkaIdempotencyConfig` with a mapper that sets the `idempotency(...)` automatically when `.build()` is called.

Similar for `.correlation()` to return a `HttpKafkaCorrelationConfig` with a mapper that sets the `correlation(...)` automatically when `.build()` is called.

We shouldn't need to create separate sub builders in the adapter.